pywinauto/controls/uia_controls.py

```diff
@@ -937,9 +946,23 @@ def button(self, button_identifier, exact=True):
         * **exact** flag specifies if the exact match for the text look up
           has to be applied.
         """
+        if self.win32_wrapper is not None:
+            btn_count = self.win32_wrapper.button_count()
+            cc = []
+            cc_texts = set()
+            for btn_num in range(btn_count):
+                button_coord_x, button_coord_y = self.client_to_screen(self.win32_wrapper.
+                                                                       get_button_rect(btn_num).mid_point())
+                btn_elem_info = UIAElementInfo.from_point(button_coord_x, button_coord_y)
+                ctrl_text = btn_elem_info.name
+                if ctrl_text not in cc_texts:
+                    cc_texts.add(ctrl_text)
```

Comment:
`cc_texts` is not used except checking to be unique which is redundant. Please `append` in any case and remove `cc_texts`.